common/com/twitter/intellij/pants/util/PantsUtil.java

```diff
@@ -188,6 +190,16 @@ public static boolean isPantsProjectFile(VirtualFile file) {
     return isBUILDFileName(file.getName());
   }
 
+  /**
+   * Almost exactly like isPantsProjectFile, but checks that the BUILD
+   * file is under a Pants project.
+   *
+   * @param file - a BUILD file
+   */
+  public static boolean isFileUnderPantsProject(VirtualFile file) {
+    return findPantsExecutable(file).isPresent() && isBUILDFileName(file.getName());
```

Comment:
`&& isBUILDFileName(file.getName()` should be unnecessary because later it checks for `PantsUtil.isBUILDFileName(file.getName())` explicitly.

Reply:
nit: `isFileUnderPantsProject` -> `isFileUnderPantsRepo`